easybuild/easyblocks/m/mxnet.py

```diff
@@ -0,0 +1,189 @@
+                break
+
+        if self.mxnet_src_dir is None:
+            raise EasyBuildError("Could not find the MXNet source directory")
+
+        self.log.debug("MXNet dir is: %s", self.mxnet_src_dir)
+
+        for srcdir in os.listdir(self.builddir):
+            if srcdir.startswith("mxnet-"):
+                continue
+            else:
+                submodule, _, _ = srcdir.rpartition('-')
+                try:
+                    newdir = os.path.join(self.mxnet_src_dir, submodule)
+                    olddir = os.path.join(self.builddir, srcdir)
+                    # first remove empty existing directory
+                    os.rmdir(newdir)
+                    shutil.move(olddir, newdir)
+                except IOError, err:
+                    raise EasyBuildError("Failed to move %s to %s: %s", olddir, newdir, err)
+
+        # the nnvm submodules has dmlc-core as a submodule too. Let's up a symlink in place.
+        try:
+            newdir = os.path.join(self.mxnet_src_dir, "nnvm", "dmlc-core")
+            olddir = os.path.join(self.mxnet_src_dir, "dmlc-core")
+            os.rmdir(newdir)
+            symlink(olddir, newdir)
+        except IOError, err:
+            raise EasyBuildError("Failed to symlink %s to %s: %s", olddir, newdir, err)
+
+    def configure_step(self):
+        """Patch 'config.mk' file to use EB stuff"""
+        copy_file('make/config.mk', '.')
+
+        regex_subs = [
+            (r"export CC = gcc", r"# \g<0>"),
+            (r"export CXX = g\+\+", r"# \g<0>"),
+            (r"(?P<var>ADD_LDFLAGS\s*=)\s*$", r"\g<var> %s" % os.environ['EBVARCFLAGS']),
+            (r"(?P<var>ADD_CFLAGS\s*=)\s*$", r"\g<var> %s" % os.environ['EBVARLDFLAGS']),
+        ]
+
+        toolchain_blas = self.toolchain.definition().get('BLAS', None)[0]
+        if toolchain_blas == 'imkl':
+            blas = "mkl"
+            imkl_version = get_software_version('imkl')
+            if LooseVersion(imkl_version) >= LooseVersion('17'):
+                regex_subs.append(("USE_MKL2017 = 0", "USE_MKL2017 = 1"))
+            regex_subs.append((r"(?P<var>MKLML_ROOT=).*$", r"# \g<var>%s" % os.environ["MKLROOT"]))
+        elif toolchain_blas in ['ACML', 'ATLAS']:
+            blas = "atlas"
+        elif toolchain_blas == 'OpenBLAS':
+            blas = "openblas"
+        elif toolchain_blas is None:
+            # This toolchain has no BLAS library
+            raise EasyBuildError("No BLAS library found in the toolchain")
+
+        regex_subs.append((r'USE_BLAS =.*', 'USE_BLAS = %s' % blas))
+
+        if get_software_root('NNPACK'):
+            regex_subs.append(("USE_NNPACK = 0", "USE_NNPACK = 1"))
+
+        apply_regex_substitutions('config.mk', regex_subs)
+
+        super(EB_MXNet, self).configure_step()
+
+    def install_step(self):
+        """Specify list of files to copy"""
+        self.cfg['files_to_copy'] = ['bin', 'include', 'lib',
```

Comment:
I would, just to avoid surprises if people try to specify `files_to_copy` in the easyconfig (which would get bluntly ignored right now)